server.rb

```diff
@@ -6,28 +11,48 @@
 
 use Rack::Logger
 
-set :config_path, File.expand_path('../config.yml', __FILE__)
-set :config, ->{ YAML.load(ERB.new(File.read(config_path)).result) }
+configure :development do
+  require 'dotenv'
+  Dotenv.load
+end
+
+configure do
+  set :config_path, File.expand_path('../config.yml', __FILE__)
+  set :config, ->{ YAML.load(ERB.new(File.read(config_path)).result) }
+  set :secret, ENV["SECRET"]
+  puts "SECRET=#{ENV["SECRET"]}"
+  puts settings.config.inspect
+end
+
 
 helpers do
   def logger
     request.logger
   end
+
   def config
     settings.config[params['app']] || {}
   end
 end
 
+before do
+  if settings.secret && params["secret"] != settings.secret
+    halt 401, "Unauthorized"
+  end
+end
+
 get '/' do
+  content_type :json
   status 200
-  "ok"
+  JSON.pretty_generate(settings.config)
```

Comment:
i was thinking if the user didn't secure their app then we don't want to dump this

if we don't have that logic, then we should probably just require a secret. which seems reasonable.